crates/dids/src/method/key.rs

```diff
@@ -0,0 +1,124 @@
+use crate::did::Did;
+use crate::method::{DidMethod, DidMethodError, DidResolutionResult};
+use async_trait::async_trait;
+use crypto::key::{Key, KeyType};
+use crypto::key_manager::KeyManager;
+use did_method_key::DIDKey as SpruceDidKeyMethod;
+use ssi_dids::did_resolve::{DIDResolver, ResolutionInputMetadata};
+use ssi_dids::{DIDMethod, Source};
+use std::sync::Arc;
+
+/// Concrete implementation for a did:key DID
+pub struct DidKey {
+    uri: String,
+    key_manager: Arc<dyn KeyManager>,
+}
+
+pub struct DidKeyCreateOptions {
+    pub key_type: KeyType,
+}
+
+impl Did for DidKey {
+    fn uri(&self) -> &str {
+        &self.uri
+    }
+
+    fn key_manager(&self) -> &Arc<dyn KeyManager> {
+        &self.key_manager
+    }
+}
+
+#[async_trait]
+impl DidMethod<DidKey, DidKeyCreateOptions> for DidKey {
+    const NAME: &'static str = "key";
+
+    fn create(
```

Comment:
Does it make sense to add documentation to public facing functions?

Reply:
This isn't public. `DidMethod` is public, and it is documented here: https://github.com/TBD54566975/web5-rs/blob/main/crates/dids/src/method/mod.rs#L21-L42

Reply:
> This isn't public.

The implementation of `DidMethod::create` is being done via `DidKey::create`. Since the former is public, I thought that meant this method is public as well....

Anyways, I don't think it's important. My main concern is that there is no place in documentation that describes to consumers of the crate how to use `DidKeyCreateOptions` when creating a did of type key. While this is obvious for `DidKey`, it will become not so obvious for `DidDht` in the future.

Reply:
We can address this when we implement `DidDht`